src/LightningUserWallet.ts

```diff
@@ -44,8 +44,18 @@ export class LightningUserWallet extends OnChainMixin(LightningMixin(UserWallet)
         )
 
         if (userPastState.earn.findIndex((item) => item === id) === -1) {
-          const invoice = await this.addInvoice({ memo: id, value: amount })
-          await lightningFundingWallet.pay({ invoice, isReward: true })
+          if (this.user.username) {
```

Comment:
so this optimization would benefits only if the user already have set his username. which my guess (we could look at the data, don't have yet a practical way to do it) is that this would generally not the case

instead of a username, the pay function could accept a uid. this would solve this particular issue. this would be a bigger refactor.

but now the bigger question is whether this invoice creating is a main contribution to the lock issue, or really this is something else.

Reply:
the issue for instance could be that redis lock takes too long to acquire for a high throughput, and the issue arise where there are too many users doing the quiz at the same time.